File: mudu_api/csharp/uni/UniQueryArgv.cs
namespace Universal {

using MessagePack;
using MessagePack.Formatters;
using System.Collections.Generic;




[MessagePackObject]
public struct UniQueryArgv {
    
    [global::System.Diagnostics.CodeAnalysis.SetsRequiredMembers]
    public UniQueryArgv()
    {
        
        Oid = new UniOid();
        
        Query = new UniSqlStmt();
        
        ParamList = new UniSqlParam();
        
    }
    
    
    
    [Key(0)]
    public required UniOid Oid { get; set; }
    
    
    [Key(1)]
    public required UniSqlStmt Query { get; set; }
    
    
    [Key(2)]
    public required UniSqlParam ParamList { get; set; }
    
}

}

File: mudu_binding/src/universal/uni_query_argv.rs
use crate::universal::uni_oid::UniOid;

use crate::universal::uni_sql_stmt::UniSqlStmt;

use crate::universal::uni_sql_param::UniSqlParam;

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Default)]
pub struct UniQueryArgv {
    pub oid: UniOid,

    pub query: UniSqlStmt,

    pub param_list: UniSqlParam,
}
